src/main.ts

```diff
@@ -655,12 +669,16 @@ function generateEncode(ctx: Context, fullName: string, messageDesc: DescriptorP
     let writeSnippet: (place: string) => Code;
     if (isScalar(field) || isEnum(field)) {
       const tag = ((field.number << 3) | basicWireType(field.type)) >>> 0;
-      writeSnippet = (place) => code`writer.uint32(${tag}).${toReaderCall(field)}(${place})`;
-    } else if (isTimestamp(field)) {
+      writeSnippet = (place) => code`writer.uint32(${tag}).${toReaderCall(field.type)}(${place})`;
+    } else if (isTimestamp(field) && options.useDate === DateOption.DATE) {
       const tag = ((field.number << 3) | 2) >>> 0;
       const type = basicTypeName(ctx, field, { keepValueType: true });
       writeSnippet = (place) =>
         code`${type}.encode(${utils.toTimestamp}(${place}), writer.uint32(${tag}).fork()).ldelim()`;
+    } else if (isTimestamp(field) && options.useDate === DateOption.STRING) {
```

Comment:
Same as above, this needs to convert from a string to a protobuf timestamp format rather than just writing the string.